### ISSUE TITLE
Monitor initialization fails

### ISSUE THREAD

**Issue by Logan2211** [ceph/ceph-ansible#1196]
https://github.com/ceph/ceph-ansible/commit/471be5e629ecbba4be750db7837ff909d6e6a2e0 seems to have broken initialization of new clusters/monitors in my env. This is on a brand new cluster with a fresh fsid in the `fsid` var.

```
00:49:44 TASK [ceph.ceph-common : install ceph] *****************************************
00:50:24 changed: [aio1_ceph-mon_container-c8085cea] => (item=[u'ceph', u'ceph-common', u'ceph-fs-common', u'ceph-fuse', u'libcephfs1'])
00:50:24 
00:50:24 TASK [ceph.ceph-common : install ceph-test] ************************************
00:50:24 
00:50:24 TASK [ceph.ceph-common : install rados gateway] ********************************
00:50:24 
00:50:24 TASK [ceph.ceph-common : install ceph mds] *************************************
00:50:24 
00:50:24 TASK [ceph.ceph-common : install jemalloc] *************************************
00:50:24 
00:50:24 TASK [ceph.ceph-common : install NFS gateway] **********************************
00:50:24 
00:50:24 TASK [ceph.ceph-common : include] **********************************************
00:50:24 
00:50:24 TASK [ceph.ceph-common : include] **********************************************
00:50:24 
00:50:24 TASK [ceph.ceph-common : include] **********************************************
00:50:24 
00:50:24 TASK [ceph.ceph-common : get ceph version] *************************************
00:50:25 ok: [aio1_ceph-mon_container-c8085cea]
00:50:25 
00:50:25 TASK [ceph.ceph-common : is ceph running already?] *****************************
00:50:27 ok: [aio1_ceph-mon_container-c8085cea -> 172.29.238.24]
00:50:27 
00:50:27 TASK [ceph.ceph-common : create a local fetch directory if it does not exist] **
00:50:27 ok: [aio1_ceph-mon_container-c8085cea -> localhost]
00:50:27 
00:50:27 TASK [ceph.ceph-common : set_fact] *********************************************
00:50:27 
00:50:27 TASK [ceph.ceph-common : set_fact] *********************************************
00:50:28 ok: [aio1_ceph-mon_container-c8085cea]
00:50:28 
00:50:28 TASK [ceph.ceph-common : set_fact] *********************************************
00:50:28 
00:50:28 TASK [ceph.ceph-common : check if /etc/openstack_deploy/ceph-fetch/ directory exists] ***
00:50:29 ok: [aio1_ceph-mon_container-c8085cea -> localhost]
00:50:29 
00:50:29 TASK [ceph.ceph-common : check if /var/lib/ceph/mon/ceph-aio1-ceph-mon-container-c8085cea/keyring already exists] ***
00:50:30 ok: [aio1_ceph-mon_container-c8085cea]
00:50:30 
00:50:30 TASK [ceph.ceph-common : fail if /var/lib/ceph/mon/ceph-aio1-ceph-mon-container-c8085cea/keyring doesn't exist] ***
00:50:30 
00:50:30 TASK [ceph.ceph-common : get existing initial mon keyring if it already exists but not monitor_keyring.conf in /etc/openstack_deploy/ceph-fetch/] ***
00:50:30 skipping: [aio1_ceph-mon_container-c8085cea]
00:50:30 
00:50:30 TASK [ceph.ceph-common : test existing initial mon keyring] ********************
00:50:31 fatal: [aio1_ceph-mon_container-c8085cea]: FAILED! => {"changed": true, "cmd": ["ceph", "--connect-timeout", "3", "--cluster", "ceph", "--keyring", "/var/lib/ceph/mon/ceph-aio1-ceph-mon-container-c8085cea/keyring", "-n", "mon.", "fsid"], "delta": "0:00:00.069849", "end": "2016-12-16 00:50:31.842993", "failed": true, "rc": 1, "start": "2016-12-16 00:50:31.773144", "stderr": "Error initializing cluster client: Error('error calling conf_read_file: error code 22',)", "stdout": "", "stdout_lines": [], "warnings": []}
00:50:31 ...ignoring
00:50:31 
00:50:31 TASK [ceph.ceph-common : fail if initial mon keyring found doesn't work] *******
00:50:32 fatal: [aio1_ceph-mon_container-c8085cea]: FAILED! => {"changed": false, "failed": true, "msg": "Initial mon keyring found doesn't work."}
```

**Comment by leseb** [comment]
It looks like Ansible has found a keyring at `/var/lib/ceph/mon/ceph-aio1-ceph-mon-container-c8085cea/keyring`, later it tried to test that key but didn't succeed.
I believe you had some leftover on that machine, can you confirm us that or give us a bit more details about your env?
Thanks!

**Comment by AlBully** [comment]
We have the same problem.
Here a little more verbose output:

###################################
TASK [ceph.ceph-common : test existing initial mon keyring] ********************
task path: /root/ceph-ansible/ceph-ansible-vmtest-debian-jewel-xfs/roles/ceph-common/tasks/facts.yml:63
<vm-ceph-001> ESTABLISH SSH CONNECTION FOR USER: ansible
<vm-ceph-001> SSH: EXEC ssh -C -q -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=ansible -o ConnectTimeout=10 -o ControlPath=/root/.ansible/cp/ansible-ssh-%h-%p-%r vm-ceph-001 '/bin/sh -c '"'"'( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1481897443.94-107275925544164 `" && echo ansible-tmp-1481897443.94-107275925544164="` echo $HOME/.ansible/tmp/ansible-tmp-1481897443.94-107275925544164 `" ) && sleep 0'"'"''
<vm-ceph-001> PUT /tmp/tmp6lBR9c TO /home/ansible/.ansible/tmp/ansible-tmp-1481897443.94-107275925544164/command
<vm-ceph-001> SSH: EXEC sftp -b - -C -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=ansible -o ConnectTimeout=10 -o ControlPath=/root/.ansible/cp/ansible-ssh-%h-%p-%r '[vm-ceph-001]'
<vm-ceph-001> ESTABLISH SSH CONNECTION FOR USER: ansible
<vm-ceph-001> SSH: EXEC ssh -C -q -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=ansible -o ConnectTimeout=10 -o ControlPath=/root/.ansible/cp/ansible-ssh-%h-%p-%r -tt vm-ceph-001 '/bin/sh -c '"'"'sudo -H -S -n -u root /bin/sh -c '"'"'"'"'"'"'"'"'echo BECOME-SUCCESS-yzsahlrlystjzhmrrofgxbqvwqicycev; LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 /usr/bin/python /home/ansible/.ansible/tmp/ansible-tmp-1481897443.94-107275925544164/command; rm -rf "/home/ansible/.ansible/tmp/ansible-tmp-1481897443.94-107275925544164/" > /dev/null 2>&1'"'"'"'"'"'"'"'"' && sleep 0'"'"''
fatal: [vm-ceph-001]: FAILED! => {"changed": true, "cmd": ["ceph", "--connect-timeout", "3", "--cluster", "ceph", "--keyring", "/var/lib/ceph/mon/ceph-vm-ceph-001/keyring", "-n", "mon.", "fsid"], "delta": "0:00:00.042995", "end": "2016-12-16 14:10:44.078571", "failed": true, "invocation": {"module_args": {"_raw_params": "ceph --connect-timeout 3 --cluster ceph --keyring /var/lib/ceph/mon/ceph-vm-ceph-001/keyring -n mon. fsid", "_uses_shell": false, "chdir": null, "creates": null, "executable": null, "removes": null, "warn": true}, "module_name": "command"}, "rc": 1, "start": "2016-12-16 14:10:44.035576", "stderr": "Error initializing cluster client: Error('error calling conf_read_file: error code 22',)", "stdout": "", "stdout_lines": [], "warnings": []}
...ignoring

TASK [ceph.ceph-common : fail if initial mon keyring found doesn't work] *******
task path: /root/ceph-ansible/ceph-ansible-vmtest-debian-jewel-xfs/roles/ceph-common/tasks/facts.yml:68
fatal: [vm-ceph-001]: FAILED! => {"changed": false, "failed": true, "invocation": {"module_args": {"msg": "Initial mon keyring found doesn't work."}, "module_name": "fail"}, "msg": "Initial mon keyring found doesn't work."}
#################################

So long
Marco

**Comment by Logan2211** [comment]
Nope-- completely fresh env on a VM built by my CI. I just froze one of the failed builds and am working to pull verbose debugging, will post that asap. I did check this however:
```
root@aio1-ceph-mon-container-1f9c75be:~# ls -lh /var/lib/ceph/mon
total 0
root@aio1-ceph-mon-container-1f9c75be:~# ls -lh /etc/ceph
total 4.0K
-rw-r--r-- 1 root root 92 Dec  9 20:46 rbdmap
```

Based on the fail `"Error initializing cluster client: Error('error calling conf_read_file: error code 22',)"` I suspect the issue is this command requires an existing ceph conf, but on a new, uninitialized cluster, we do not have ceph.conf even generated yet.

**Comment by Logan2211** [comment]
Verbose output http://cdn.pasteraw.com/1g6h1y834yymn2ugsp1b4ijvrhw3xme

**Comment by leseb** [comment]
Somehow you're entering the `block` when `ceph_current_fsid.rc != 0` which should not happen. This seems to be working on Ansible 2.2, what's your Ansible version?

**Comment by Logan2211** [comment]
seeing it on 2.1.1.0

**Comment by leseb** [comment]
any chance to try this on 2.2.0.0?

**Comment by Logan2211** [comment]
Seems to work on 2.2.0.0

**Comment by Logan2211** [comment]
I guess we could work around it with a task file include instead of the block functionality which appears to be broken?

**Comment by Logan2211** [comment]
Tested the fix in #1199 on 2.1.1.0 and it works for me.

**Comment by AlBully** [comment]
Upgrade to ansible 2.2.0.0 solved my problem.
